crates/assistant/src/completion_provider.rs

```diff
@@ -24,6 +24,21 @@ use settings::{Settings, SettingsStore};
 use std::sync::Arc;
 use std::time::Duration;
 
+/// Choose which model to use for openai provider based on the available models and the default model.
+fn choose_openai_model(
+    model: &::open_ai::Model,
+    available_models: &Option<Vec<::open_ai::Model>>,
+) -> ::open_ai::Model {
+    let available_models = available_models.as_ref();
+
+    // available_models is set but default model isn't in the list, use the first available model instead
+    if available_models.is_some_and(|models| !models.is_empty() && !models.contains(model)) {
+        available_models.unwrap()[0].clone()
```

Comment:
Can we rule out potential panics here (from `unwrap` and `[0]`)?

For example, by doing something like this:

```rust
available_models
    .and_then(|models| models.find(model).or_else(|| models.first()))
    .unwrap_or_else(|| model)
    .clone()
```